gpkit/nomials/nomial_math.py

```diff
@@ -346,6 +348,8 @@ def __le__(self, other):
 
     # Posynomial.__ge__ falls back on Signomial.__ge__
 
+    __hash__ = Signomial.__hash__  # required by Python 3
+
```

Comment:
interesting. Specifically,

"If a class that overrides __eq__() needs to retain the implementation of __hash__() from a parent class, the interpreter must be told this explicitly by setting __hash__ = <ParentClass>.__hash__."  (https://docs.python.org/3/reference/datamodel.html#object.__hash__)